site/source/tutorials/vdom_abstractions.md

```diff
@@ -0,0 +1,108 @@
+---
+layout: false
+title: vDOM Abstractions - w and v in the d module
+overview: Abstracting DOM allows us to create presentation components programatically. Dojo 2 provides a layer of abstraction between the DOM and its virtual counterpart as it exists in JavaScript and a layer of abstraction between this virtual DOM and Dojo 2 components - using a common syntax in a common module.
+---
+
+## `@dojo/widget-core/d.ts`
+
+Creating the DOM structure to support the syntax, style, and structure required to display a custom component may require a significant number of nodes. Because of this, many virtual DOM implementations have chosen single-letter function names to reduce typing and add clarity when reading code.
+
+This module, named for what would be the single-letter shorthand function name for "Dojo 2" contains two single-letter functions each named for their purpose:
+
+- `v`: is used to create an abstract version of a DOM node
+- `w`: is used to create an abstract version of a widget
+
+## Common variables
+
+A node can be thought of as an object of a certain type, customized through properties, dispatching events through callbacks, with zero or more children. Condensing this down into a function where each argument can be isolated gives us a type, a set of properties, and a list of children.
+
+Because Dojo 2 is built on TypeScript, splitting arguments into these parts allows static type enforcement that can be based on the type of node being created. It also helps explain why `v` and `w` are separate functions, as each function enforces a different set of constraints on its arguments.
+
+## `v` for vDOM
+
+Dojo's virtual DOM function expects to receive the tag name, properties, and children as its three arguments. Only the tag name is required and other arguments may be ommitted as long as properties and children are called in that order.
+
+```ts
+v('article');
+// represents <article />
+v('article', { id: 'first' });
+// represents <article id="first" />
+v('article', { id: 'first' }, [ v('p') ]);
+// represents <article id="first"><p /></article>
```

Comment:
Maybe we should just go with <p></p> instead of xhtml syntax (same with the li below)?